### PR TITLE
Update tomcat_wrk to report more samples.

### PR DESCRIPTION
**Pull request by cmccoy**
* Latency / throughput for connection count with minimum p50 latency (labeled with `best_p50`).
* Latency / throughput for connection count with maximum throughput (labeled with `best_throughput`).
* (optionally) Samples from all other connection counts.

### CODE REVIEW

**Comment by meteorfox** [comment]
Cool that was fast! I'll try it out.

**Comment by meteorfox** [comment]
Here's the data I collected

general1-2 comes with just 2 vCPUs, and network bandwidth is capped around 400Mbps. 

```bash
./pkb.py --cloud=Rackspace --machine_type=general1-2 --benchmarks=tomcat_wrk
```

```
-------------------------PerfKitBenchmarker Results Summary-------------------------
TOMCAT_WRK:
  duration="120" ip_type="external" runtime_in_seconds="120"
  bytes transferred              1413810000.000000 bytes                          (best_p50="True" connections="1" threads="1")
  errors                                0.000000 n                              (best_p50="True" connections="1" threads="1")
  p25 latency                           0.380000 ms                             (best_p50="True" connections="1" threads="1")
  p5 latency                            0.361000 ms                             (best_p50="True" connections="1" threads="1")
  p50 latency                           0.408000 ms                             (best_p50="True" connections="1" threads="1")
  p75 latency                           0.497000 ms                             (best_p50="True" connections="1" threads="1")
  p90 latency                           2.865000 ms                             (best_p50="True" connections="1" threads="1")
  p99 latency                         162.675000 ms                             (best_p50="True" connections="1" threads="1")
  p99.9 latency                       650.301000 ms                             (best_p50="True" connections="1" threads="1")
  requests                         229955.000000 n                              (best_p50="True" connections="1" threads="1")
  throughput                         1915.860000 requests/sec                   (best_p50="True" connections="1" threads="1")
  bytes transferred              1496350000.000000 bytes                          (best_throughput="True" connections="128" threads="2")
  errors                                0.000000 n                              (best_throughput="True" connections="128" threads="2")
  p25 latency                          51.074000 ms                             (best_throughput="True" connections="128" threads="2")
  p5 latency                           29.588000 ms                             (best_throughput="True" connections="128" threads="2")
  p50 latency                          62.624000 ms                             (best_throughput="True" connections="128" threads="2")
  p75 latency                          69.940000 ms                             (best_throughput="True" connections="128" threads="2")
  p90 latency                          77.647000 ms                             (best_throughput="True" connections="128" threads="2")
  p99 latency                         358.331000 ms                             (best_throughput="True" connections="128" threads="2")
  p99.9 latency                       571.031000 ms                             (best_throughput="True" connections="128" threads="2")
  requests                         243356.000000 n                              (best_throughput="True" connections="128" threads="2")
  throughput                         2026.390000 requests/sec                   (best_throughput="True" connections="128" threads="2")
  End to End Runtime                 1201.710473 seconds
```

Very interesting, the relative improvement between the best median latency vs. the max throughput from a **throughput is only a 5% improvement** but the **median latency is 2x orders of magnitude worse** surprisingly the max latency actually improved, but again these are the results of just 1 run.

Anyway, the results LGTM. I'm running a second test with the flag for detailed sampels.

**Comment by meteorfox** [comment]
Alright, here a gist with all the samples: https://gist.github.com/meteorfox/49fe0ddb2a537f34f652

Same behavior as before with this choice of machine type and the nature of the benchmark, adding more connections barely improves throughput while latency severly suffers, which probably indicates that for this machine type the server is already CPU constrained.

:+1: Change LGTM, thanks for the quick turn around. I'll wait for somebody else to CR

**Comment by cmccoy** [comment]
Thanks, @meteorfox!

File: perfkitbenchmarker/linux_benchmarks/tomcat_wrk_benchmark.py [hunk 1]
```diff
@@ -155,25 +161,39 @@ def Run(benchmark_spec):
...
     logging.info('Ran with %d connections; %.2f%% errors, %.2f req/s',
-                 connections, error_rate, throughput.value)
+                 connections, error_rate, throughput)
 
     # Retry with double the connections
     connections *= 2
 
-  samples.append(max_throughput)
+  if not all_by_metric:
+    logging.warn('No requests succeeded.')
```
Comment (ehankland) [inline perfkitbenchmarker/linux_benchmarks/tomcat_wrk_benchmark.py]: nit: seems like an error to me

**Comment by ehankland** [comment]
LGTM

File: perfkitbenchmarker/linux_benchmarks/tomcat_wrk_benchmark.py (hunk 1 above)
Comment (cmccoy) [inline perfkitbenchmarker/linux_benchmarks/tomcat_wrk_benchmark.py]: Done.

**Pull request merged**